File: packages/py-sdk/examples/turbosign_send_simple.py
"""
Example 1: Send Signature Directly - Template Anchors

This example sends a document directly to recipients for signature.
Uses template anchors like {signature1} and {date1} in your PDF.

Use this when: You want to send immediately without review
"""

import asyncio
import os
from turbodocx_sdk import TurboSign

async def send_directly_example():
    # Configure TurboSign
    TurboSign.configure(
        api_key=os.getenv("TURBODOCX_API_KEY", "your-api-key-here"),
        org_id=os.getenv("TURBODOCX_ORG_ID", "your-org-id-here"),
        sender_email=os.getenv("TURBODOCX_SENDER_EMAIL", "support@yourcompany.com"),
        sender_name=os.getenv("TURBODOCX_SENDER_NAME", "Your Company Name")
    )

    try:
        # Read PDF file
        with open("../../ExampleAssets/sample-contract.pdf", "rb") as f:
            pdf_file = f.read()

        print("Sending document directly to recipients...\n")

        result = await TurboSign.send_signature(
            file=pdf_file,
            document_name="Partnership Agreement",
            document_description="Q1 2025 Partnership Agreement - Please review and sign",
            recipients=[
                {
                    "name": "John Doe",
                    "email": "john@example.com",
                    "signingOrder": 1
                },
                {
                    "name": "Jane Smith",
                    "email": "jane@example.com",
                    "signingOrder": 2
                }
            ],
            fields=[
                # First recipient's fields - using template anchors
                {
                    "type": "full_name",
                    "recipientEmail": "john@example.com",
                    "template": {
                        "anchor": "{name1}",
                        "placement": "replace",
                        "size": {"width": 100, "height": 30}
                    }
                },
                {
                    "type": "signature",
                    "recipientEmail": "john@example.com",
                    "template": {
                        "anchor": "{signature1}",       # Text in your PDF to replace
                        "placement": "replace",          # Replace the anchor text
                        "size": {"width": 100, "height": 30}
                    }
                },
                {
                    "type": "date",
                    "recipientEmail": "john@example.com",
                    "template": {
                        "anchor": "{date1}",
                        "placement": "replace",
                        "size": {"width": 75, "height": 30}
                    }
                },
                # Second recipient's fields
                {
                    "type": "full_name",
                    "recipientEmail": "jane@example.com",
                    "template": {
                        "anchor": "{name2}",
                        "placement": "replace",
                        "size": {"width": 100, "height": 30}
                    }
                },
                {
                    "type": "signature",
                    "recipientEmail": "jane@example.com",
                    "template": {
                        "anchor": "{signature2}",
                        "placement": "replace",
                        "size": {"width": 100, "height": 30}
                    }
                },
                {
                    "type": "date",
                    "recipientEmail": "jane@example.com",
                    "template": {
                        "anchor": "{date2}",
                        "placement": "replace",
                        "size": {"width": 75, "height": 30}
                    }
                }
            ]
        )

        print("✅ Document sent successfully!\n")
        print(f"Document ID: {result['documentId']}")
        print(f"Message: {result['message']}")

        # To get sign URLs and recipient details, use get_status
        try:
            status = await TurboSign.get_status(result['documentId'])
            if status.get('recipients'):
                print("\nSign URLs:")
                for recipient in status['recipients']:
                    print(f"  {recipient['name']}: {recipient.get('signUrl', 'N/A')}")
        except Exception as status_error:
            print("\nNote: Could not fetch recipient sign URLs")

    except Exception as error:
        print(f"Error: {error}")

# Run the example
if __name__ == "__main__":
    asyncio.run(send_directly_example())

File: packages/go-sdk/examples/turbosign_send_simple.go
// Example 1: Send Signature Directly - Template Anchors
//
// This example sends a document directly to recipients for signature.
// Uses template anchors like {signature1} and {date1} in your PDF.
//
// Use this when: You want to send immediately without review

package main

import (
	"context"
	"fmt"
	"os"

	turbodocx "github.com/TurboDocx/SDK/packages/go-sdk"
)

func main() {
	// Configure TurboDocx client
	client, err := turbodocx.NewClientWithConfig(turbodocx.ClientConfig{
		APIKey:      getEnv("TURBODOCX_API_KEY", "your-api-key-here"),
		OrgID:       getEnv("TURBODOCX_ORG_ID", "your-org-id-here"),
		SenderEmail: getEnv("TURBODOCX_SENDER_EMAIL", "support@yourcompany.com"),
		SenderName:  getEnv("TURBODOCX_SENDER_NAME", "Your Company Name"),
	})
	if err != nil {
		fmt.Printf("Error creating client: %v\n", err)
		return
	}

	// Read PDF file
	pdfFile, err := os.ReadFile("../../ExampleAssets/sample-contract.pdf")
	if err != nil {
		fmt.Printf("Error reading file: %v\n", err)
		return
	}

	fmt.Println("Sending document directly to recipients...\n")

	ctx := context.Background()
	result, err := client.TurboSign.SendSignature(ctx, &turbodocx.SendSignatureRequest{
		File:                pdfFile,
		FileName:            "sample-contract.pdf",
		DocumentName:        "Partnership Agreement",
		DocumentDescription: "Q1 2025 Partnership Agreement - Please review and sign",
		Recipients: []turbodocx.Recipient{
			{Name: "John Doe", Email: "john@example.com", SigningOrder: 1},
			{Name: "Jane Smith", Email: "jane@example.com", SigningOrder: 2},
		},
		Fields: []turbodocx.Field{
			// First recipient's fields - using template anchors
			{
				Type:           "full_name",
				RecipientEmail: "john@example.com",
				Template: &turbodocx.TemplateAnchor{
					Anchor:    "{name1}",
					Placement: "replace",
					Size:      &turbodocx.Size{Width: 100, Height: 30},
				},
			},
			{
				Type:           "signature",
				RecipientEmail: "john@example.com",
				Template: &turbodocx.TemplateAnchor{
					Anchor:    "{signature1}", // Text in your PDF to replace
					Placement: "replace",       // Replace the anchor text
					Size:      &turbodocx.Size{Width: 100, Height: 30},
				},
			},
			{
				Type:           "date",
				RecipientEmail: "john@example.com",
				Template: &turbodocx.TemplateAnchor{
					Anchor:    "{date1}",
					Placement: "replace",
					Size:      &turbodocx.Size{Width: 75, Height: 30},
				},
			},
			// Second recipient's fields
			{
				Type:           "full_name",
				RecipientEmail: "jane@example.com",
				Template: &turbodocx.TemplateAnchor{
					Anchor:    "{name2}",
					Placement: "replace",
					Size:      &turbodocx.Size{Width: 100, Height: 30},
				},
			},
			{
				Type:           "signature",
				RecipientEmail: "jane@example.com",
				Template: &turbodocx.TemplateAnchor{
					Anchor:    "{signature2}",
					Placement: "replace",
					Size:      &turbodocx.Size{Width: 100, Height: 30},
				},
			},
			{
				Type:           "date",
				RecipientEmail: "jane@example.com",
				Template: &turbodocx.TemplateAnchor{
					Anchor:    "{date2}",
					Placement: "replace",
					Size:      &turbodocx.Size{Width: 75, Height: 30},
				},
			},
		},
	})

	if err != nil {
		fmt.Printf("Error: %v\n", err)
		return
	}

	fmt.Println("✅ Document sent successfully!\n")
	fmt.Printf("Document ID: %s\n", result.DocumentID)
	fmt.Printf("Message: %s\n", result.Message)

	// To get sign URLs and recipient details, use GetStatus
	status, err := client.TurboSign.GetStatus(ctx, result.DocumentID)
	if err == nil && status.Recipients != nil {
		fmt.Println("\nSign URLs:")
		for _, recipient := range status.Recipients {
			fmt.Printf("  %s: %s\n", recipient.Name, recipient.SignURL)
		}
	} else {
		fmt.Println("\nNote: Could not fetch recipient sign URLs")
	}
}

func getEnv(key, fallback string) string {
	if value := os.Getenv(key); value != "" {
		return value
	}
	return fallback
}
